tests/tck/features/optimizer/EmbedEdgeAllPredIntoTraverseRule.feature

```diff
@@ -0,0 +1,284 @@
+      | id | name           | dependencies | profiling data | operator info                    |
+      | 7  | Project        | 11           |                |                                  |
+      | 11 | AppendVertices | 13           |                |                                  |
+      | 13 | Traverse       | 1            |                | {"filter": "(like.likeness>90)"} |
+      | 1  | IndexScan      | 2            |                |                                  |
+      | 2  | Start          |              |                |                                  |
+    When profiling query:
+      """
+      MATCH (v:player)-[e:like*0..5]->(n)
+      WHERE all(i in e where i.likeness>90) AND size(e)>0 AND (n.player.age>0) == true
+      RETURN [i in e | i.likeness] AS likeness, n.player.age AS nage
+      """
+    Then the result should be, in any order:
+      | likeness         | nage |
+      | [99, 99]         | 34   |
+      | [99]             | 38   |
+      | [99, 99]         | 38   |
+      | [99]             | 34   |
+      | [100]            | 43   |
+      | [95, 95, 95]     | 41   |
+      | [95, 95]         | 42   |
+      | [95, 95]         | 41   |
+      | [95]             | 36   |
+      | [95]             | 41   |
+      | [95]             | 30   |
+      | [100]            | 31   |
+      | [95, 95, 95]     | 41   |
+      | [95, 95]         | 36   |
+      | [95, 95]         | 41   |
+      | [95]             | 42   |
+      | [95]             | 41   |
+      | [99, 95, 95, 95] | 41   |
+      | [99, 95, 95, 95] | 41   |
+      | [99, 95, 95]     | 36   |
+      | [99, 95, 95]     | 41   |
+      | [99, 95, 95]     | 42   |
+      | [99, 95, 95]     | 41   |
+      | [99, 95]         | 42   |
+      | [99, 95]         | 41   |
+      | [99, 95]         | 36   |
+      | [99, 95]         | 41   |
+      | [99, 100]        | 43   |
+      | [99]             | 36   |
+      | [99]             | 42   |
+      | [99]             | 30   |
+      | [99]             | 32   |
+      | [99]             | 41   |
+      | [99]             | 34   |
+      | [99]             | 25   |
+      | [99]             | 30   |
+      | [99]             | 29   |
+      | [99]             | 31   |
+      | [99]             | 33   |
+    And the execution plan should be:
+      | id | name           | dependencies | profiling data | operator info                             |
+      | 7  | Project        | 15           |                |                                           |
+      | 15 | Filter         | 11           |                | {"condition": "((n.player.age>0)==true)"} |
+      | 11 | AppendVertices | 14           |                |                                           |
+      | 14 | Filter         | 13           |                | {"condition": "(size($e)>0)"}             |
+      | 13 | Traverse       | 1            |                | {"edge filter": "(*.likeness>90)"}        |
```

Comment:
why the predicate is sometimes in `edge filter` and sometimes in `filter` fields of `Traverse`?

Reply:
0 step filter cannot push to storage layer.